### PR TITLE
DP0001, in-memory database setup

### PR DESCRIPTION
**Pull request by d-pilipavicius**
Added create business functionality to the APIs (Task DP0001).
This required the creation of BusinessComponent.
A H2 in-memory database is used to hold data, though it is not initialized.
The testing of the API requires to manually insert a User to the table in the database.

### CODE REVIEW

File: src/main/java/com/example/demo/BusinessComponent/Domain/Services/Interfaces/IBusinessService.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.example.demo.BusinessComponent.Domain.Services.Interfaces;
+
+import com.example.demo.BusinessComponent.API.DTOs.BusinessDTO;
+import com.example.demo.BusinessComponent.API.DTOs.CreateBusinessDTO;
+
+public interface IBusinessService {
+  public BusinessDTO createBusiness(CreateBusinessDTO dto);
+}
```
Comment (xDziugas) [inline src/main/java/com/example/demo/BusinessComponent/Domain/Services/Interfaces/IBusinessService.java]: IBusinessService will always have 1 implementation, no need to use interface here.

File: src/main/java/com/example/demo/BusinessComponent/Helpers/Mappers/Interfaces/IBusinessMapper.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.example.demo.BusinessComponent.Helpers.Mappers.Interfaces;
+
+import com.example.demo.BusinessComponent.API.DTOs.BusinessDTO;
+import com.example.demo.BusinessComponent.API.DTOs.CreateBusinessDTO;
+import com.example.demo.BusinessComponent.Domain.Entities.Business;
+
+public interface IBusinessMapper {
+  public BusinessDTO businessToBusinessDTO(Business business);
+  public Business createBusinessDTOToBusiness(CreateBusinessDTO createBusinessDTO);
+}
```
Comment (xDziugas) [inline src/main/java/com/example/demo/BusinessComponent/Helpers/Mappers/Interfaces/IBusinessMapper.java]: IBusinessMapper will also have only 1 implementation, consider implementing a more generic solution if needed such as IMapper or similar

**Review by VarArnas — DISMISSED** [review]
Man atrodo all good, tik tikriausiai reiketu vieno folderio in repositories interfacams, bet cia aisku ir veliau galima. Taip pat jei darom interfac'us ir services ir repositories, tai gal tada ir application services reikia interfaces? Nes tada controlleryje butu galima naudoti interface object vietoj class object, tai daugiau abstrakcijos butu. Bet cia siaip ideas. Krc zjbs

File: src/main/java/com/example/demo/BusinessComponent/Helpers/Mappers/BusinessMapper.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      .name(business.getName())
+      .ownerId(business.getOwner().getId())
+      .address(business.getAddress())
+      .phoneNumber(business.getPhoneNumber())
+      .emailAddress(business.getEmailAddress())
+      .build();
+  }
+
+  @Override
+  public Business createBusinessDTOToBusiness(CreateBusinessDTO createBusinessDTO) {
```
Comment (xDziugas) [inline src/main/java/com/example/demo/BusinessComponent/Helpers/Mappers/BusinessMapper.java]: optional, but i feel like if you are already using a BusinessMapper, methods could have more straightforward names such as modelToDTO or mapToDTO

File: src/main/java/com/example/demo/BusinessComponent/Domain/Entities/User.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  @Id
+  @GeneratedValue(strategy = GenerationType.UUID)
+  private UUID id;
+
+  @Column(name = "full_name", nullable = false)
+  private String fullName;
+
+  @OneToOne
+  @JoinColumn(name = "business_id", referencedColumnName = "id", nullable = true)
+  private Business business;
```
Comment (xDziugas) [inline src/main/java/com/example/demo/BusinessComponent/Domain/Entities/User.java:43]: if Business object isnt necessary consider adding @OneToOne(fetch = FetchType.LAZY) or private UUID businessId;

**Review by xDziugas — DISMISSED** [review]

File: src/main/java/com/example/demo/BusinessComponent/Helpers/Mappers/BusinessMapper.java (hunk 3 above)
Comment (VarArnas) [reply src/main/java/com/example/demo/BusinessComponent/Helpers/Mappers/BusinessMapper.java]: Gal jei mapper'is jau pavadintas business, uztenka naudot POST/GET/PATCH ir etc. nes jau business mapper turi business name? Aij nezinau

**Review by VarArnas — APPROVED** [review]
LGTM!

**Review by xDziugas — APPROVED** [review]

**Pull request merged**